### ISSUE TITLE
Integrating NopStarterKit.Web

### ISSUE THREAD

**Issue by Willem676** [yawkale/nop-umbraco-integration#7]
Hello,

When I edit the connectionstring in NopStarterKit.Web to an empty database, the umbraco installation keeps looping and doesn't get executed.

Can this solution be started or is it just an example of how you should integrate it in your own project?

Greetings,

Willem

**Comment by mpwebstudio** [comment]
Same issue here :(

Cheers
Mike

**Comment by Willem676** [comment]
After reseaching more about this project i noticed only the product / categories / basket are currently integrated for the most part. The checkout process seems to be implementented static.

**Comment by yawkale** [comment]
For run NopStarterKit.Web you should use sample DB, it's will not work with empty DB.
Umbraco DB: http://www.nopintegra.com/forum/umbraco.zip
Nop DB: http://www.nopintegra.com/forum/nop.zip

**Comment by wmlockhart** [comment]
Has anyone got this to work?

I used the Umbraco DB from here: http://www.nopintegra.com/forum/umbraco.zip as recommended previously.

However, when I run the solution, I get the exception "The remote server returned an error: (400) Bad Request."

at Line 25: var response = request.GetResponse();

in the file: AccessProvider.cs

in the project: Nop.Api.Adapter project

I guess it's because the username and password fields are empty.

I don't know how to fix this issue.

Can someone help please?

**Comment by yawkale** [comment]
Yes, you need to define clientid and client secret on Umbraco and nop sides
you can see it here
https://github.com/yawkale/nop-umbraco-integration

**Comment by wmlockhart** [comment]
Thank you.

**Comment by wmlockhart** [comment]
I followed those instructions and it now works.

However, the product image for 'Adobe Photoshop CS4' and a number of other images, do not display in the Umbraco site for some reason.

Can you help me please?

**Comment by umairwaheed2244** [comment]
I am facing some issues, I think that system is integrated and it's opening the site which is connected but when a product or news is clicked on an error I am facing this issue
Description: An error occurred during the compilation of a resource required to service this request. Please review the following specific error details and modify your source code appropriately.
 <div class="product-image-container">
Line 14:                     <img class="product-image" src="@Model.Content.Photos.FirstOrDefault().Url" />
Line 15:                 </div>

Compiler Error Message: CS1061: 'Umbraco.Web.PublishedContentModels.Product' does not contain a definition for 'Photos' and no extension method 'Photos' accepting a first argument of type 'Umbraco.Web.PublishedContentModels.Product' could be found (are you missing a using directive or an assembly reference?).

**Comment by umairwaheed2244** [comment]
Hi i am having the same issue when using the sample umbracoDb .I had set the client id and client secret from nopsite but it is still giving The remote server returned an error: (400) Bad Request.

**Comment by yawkale** [comment]
checking this issue

**Comment by umairwaheed2244** [comment]
Hi
My nop site is configured on http://localhost:90 and UmbracoStarterKit site is configured on http://localhost:92. I have generated the client id and Client Secret keys on Nop site and configured those configurations in UmbracoStarterKit of WebConfig. like 
  <add key="ClientId" value="caedb2a9-b420-4adc-be0f-3d99b90fa069" />
    <add key="ClientSecret" value="dee7f617-bdbf-4a9e-ad90-96f00bbf7d13" />
    <add key="ServerUrl" value="http://localhost:90" />
    <add key="RedirectUrl" value="http://matrix6:93/umbraco/surface/Authorization/GetAccessToken" />
when i run the application giving an error of  "The remote server returned an error: (400) Bad Request".
Is there anyone who resolved my issues.

Thanks

**Comment by yawkale** [comment]
please verify that Callback Url in nop commerce api also configured to http://localhost:92/umbraco/surface/Authorization/GetAccessToken

**Comment by umairwaheed2244** [comment]
@yawkale  yes
 it is configured with http://localhost:92/umbraco/surface/Authorization/GetAccessToken

**Comment by umairwaheed2244** [comment]
@yawkale  Thanks.
Sometimes it works fine and sometimes displays the error  "The remote server returned an error: (400) Bad Request".  when it works fine the images are not displayed and when clicked on product item in front end it gives an error of compilations. like

Compiler Error Message: CS1061: 
'Umbraco.Web.PublishedContentModels.Product' does not contain a definition for 'Photos' and no extension method 'Photos' accepting a first argument of type 'Umbraco.Web.PublishedContentModels.
Product' could be found (are you missing a using directive or an assembly reference?)

I am in trouble so Please resolved mine issues.
Thanks

**Comment by umairwaheed2244** [comment]
@yawkale  Thanks for your efforts.
Hope you will upload the files soon and kindly let me know while you will do this.
Thanks

**Comment by yawkale** [comment]
I merge all code to master, please let me know if you still have issue with it

**Comment by umairwaheed2244** [comment]
Hi @yawkale,
Sorry to disturb you but it would be great if you could please guide me for some the points mentioned here
- Shopping cart giving error while clicking it
- My Nop Dashboard not showing anything 
- Product price and color not manageable from Umbraco, how can i manage this in Umbraco?

**Comment by yawkale** [comment]
Sorting by filters are not working - new issue created 
Shopping cart giving error while clicking it - can you show me the error?

**Comment by umairwaheed2244** [comment]
Dear @yawkale 
Error image url here you investigate further.

https://user-images.githubusercontent.com/39156395/40291828-db3d62bc-5ce0-11e8-8c07-3075046e0456.JPG

Thank You

**Comment by yawkale** [comment]
this is a complex product, simple product works for you? 

can you show what error return server  http://prntscr.com/jktlpz - you can see it as a response from ajax call

**Comment by umairwaheed2244** [comment]
Hi @yawkale actually I don't know about the complex and simple products. These errors are appearing on my console, please have a look.


https://user-images.githubusercontent.com/39156395/40344384-fbf4052e-5dad-11e8-904a-b0f980d288c0.png
https://user-images.githubusercontent.com/39156395/40344399-1134f056-5dae-11e8-8229-8a562ac95388.png

**Comment by umairwaheed2244** [comment]
Hi @yawkale 

I am in trouble so kindly suggest me or resolved my error.i have appeared issues in my project which i told you in previous comments.

Thanks

**Comment by umairwaheed2244** [comment]
@yawkale i will highly appreciate any early help as i am facing some serious problems regarding this project.

**Comment by RobertCarlaw** [comment]
I've not looked at this project for a while, but regarding the screenshots. You need to turn errors on or we've no chance in understanding what is going on. From my understanding this is fairly young and you'll need to debug/ fix/ extend as you go. 

Stick a break point, step through and let us know what the error is. 

What version of Umbraco are you using?

**Comment by yawkale** [comment]
Good idea to turn off custom errors and output of ajax call - http://prntscr.com/jr249d
Looks like some error occurs on Umbraco side or connecting to nop.
We can try to do the session on skype, my skype is yyworm

**Comment by RobertCarlaw** [comment]
You need to make sure you've got a OAuth token if I remember correctly.
